## Goal 
Filter non-continuous stocks. That is, filter out stocks that do not span the entire time series.

In [1]:
import sys
sys.path.append(r'../models/')

import pandas as pd
from data_cleaning import MARKET_DATA_PATH, NEWS_DATA_PATH, clean_market_data

In [2]:
market_train_df = pd.read_csv(MARKET_DATA_PATH)
# news_train_df = pd.read_csv(NEWS_DATA_PATH)

In [3]:
clean_market_df = clean_market_data(market_train_df)

In [5]:
clean_market_df.head(5)

assetCode       time      volume    open  returnsOpenPrevMktres1  \
14290       A.N 2007-02-15   4095135.0  32.990               -0.001572   
14291     AAI.N 2007-02-15   1378650.0  11.650               -0.001498   
14292     AAP.N 2007-02-15   3884400.0  37.000               -0.019388   
14293    AAPL.O 2007-02-15  12997017.0  85.310                0.000738   
14294     ABB.N 2007-02-15  10168100.0  18.245               -0.026040   

       returnsOpenPrevMktres10  returnsOpenNextMktres10  
14290                 0.007461                -0.029993  
14291                -0.010884                -0.013111  
14292                -0.026448                -0.028244  
14293                -0.044809                -0.014505  
14294                -0.010927                 0.017172

In [16]:
clean_market_df.time.nunique()

2488

In [10]:
clean_market_df.groupby('assetCode').size().head()

assetCode
A.N         2488
AAI.N        879
AAL.O        772
AAMRQ.OB      70
AAN.N       1486
dtype: int64

In [19]:
sel = clean_market_df.groupby('assetCode').size() == clean_market_df.time.nunique()
sum(sel)

522

522 stocks span the entire time series. We want to filter only those out.

---

In [21]:
series_len = clean_market_df.time.nunique()
clean_market_df = clean_market_df.groupby('assetCode').filter(lambda x: len(x) == series_len)

In [25]:
# Confirm that we are left only with the ones covering the whole series
assert (clean_market_df.groupby('assetCode').size() == series_len).all()